narwhal/primary/src/aggregators.rs

```diff
@@ -57,11 +64,36 @@ impl VotesAggregator {
 
         if self.weight >= committee.quorum_threshold() {
             self.weight = 0; // Ensures quorum is only reached once.
-            return Ok(Some(Certificate::new(
-                committee,
-                header.clone(),
-                self.votes.clone(),
-            )?));
+            let cert = Certificate::new(committee, header.clone(), self.votes.clone())?;
+            let (_, pks) = cert.signed_by(committee);
+
+            let certificate_digest: Digest<{ crypto::DIGEST_LENGTH }> = Digest::from(cert.digest());
+            match cert
+                .aggregated_signature
+                .verify_secure(&to_intent_message(certificate_digest), &pks[..])
+            {
+                Err(_) => {
+                    warn!(
+                        "Failed to verify aggregated sig on certificate: {}",
+                        certificate_digest
+                    );
+                    for (i, (pk, sig)) in self.votes.clone().iter().enumerate() {
+                        if sig
+                            .verify_secure(&to_intent_message(certificate_digest), pk)
+                            .is_err()
+                        {
+                            warn!(
+                                "Invalid signature on header from authority: {}",
+                                pk.encode_base64()
+                            );
+                            self.votes.remove(i);
+                            self.weight -= committee.stake(pk);
+                        }
+                    }
+                    return Ok(None);
```

Comment:
https://github.com/MystenLabs/sui/blob/b21e75596795b328cbaff39be244ab5be23bf3dd/narwhal/primary/src/certifier.rs#L285-L299 returning a None here will have the loop continue to find new votes.